File: Java/exercicio14.java
class Configuracao {
    private static Configuracao instance;

    private Configuracao() {}

    public static Configuracao getInstance() {
        if (instance == null) {
            instance = new Configuracao();
        }
        return instance;
    }
}

public class Main {
    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstance();
        Configuracao config2 = Configuracao.getInstance();
        System.out.println(config1 == config2);  // true
    }
}

File: Go/exercicio14.go
package main

import "sync"

type Configuracao struct{}

var instance *Configuracao
var once sync.Once

func GetConfiguracao() *Configuracao {
    once.Do(func() {
        instance = &Configuracao{}
    })
    return instance
}

func main() {
    config1 := GetConfiguracao()
    config2 := GetConfiguracao()
    fmt.Println(config1 == config2)  // true
}
